File: Bronze/2015/December/02_speeding_ticket/SpeedingTicket.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.PrintWriter;
import java.util.StringTokenizer;

/**
 * @author Red
 * @version 1.0
 */
public class SpeedingTicket {
    public static void main(String[] args) throws Exception {
        BufferedReader r = new BufferedReader(new FileReader("speeding.in"));
        PrintWriter pw = new PrintWriter("speeding.out");
        StringTokenizer st = new StringTokenizer(r.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        // 记录每单位距离的限速 单位为 1 mile
        int[] speedLimit = new int[100];
        int start = 0;
        for(int i = 0; i < n; i++) {
            st = new StringTokenizer(r.readLine());
            int distance = Integer.parseInt(st.nextToken());
            int speed = Integer.parseInt(st.nextToken());
            for(int j = start; j < start + distance; j++) {
                speedLimit[j] = speed;
            }
            start += distance;
        }
        // 记录每单位距离的实际速度 单位为 1 mile
        int[] realSpeed = new int[100];
        start = 0;
        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(r.readLine());
            int distance = Integer.parseInt(st.nextToken());
            int speed = Integer.parseInt(st.nextToken());
            for(int j = start; j < start + distance; j++) {
                realSpeed[j] = speed;
            }
            start += distance;
        }
        // 以 1 mile 为单位计算最大超速值
        int max = 0;
        for(int i=0;i<100;i++){
            if(realSpeed[i]-speedLimit[i]>max){
                max=realSpeed[i]-speedLimit[i];
            }
        }
        pw.println(max);
        pw.close();
    }
}

File: Bronze/2015/December/02_speeding_ticket/speeding_ticket.cpp
#include <iostream>
#include <vector>
using namespace std;

const int LEN = 100;

int main() {
    freopen("speeding.in", "r", stdin);
    freopen("speeding.out", "w", stdout);

    int N, M;
    cin >> N >> M;

    int start = 0;
    // 记录每单位距离的限速 单位为 1 mile
    vector<int> speed_limit(LEN);
    for (int i = 0; i < N; i++) {
        int distance, speed;
        cin >> distance >> speed;
        for (int j = start; j < start + distance; j++) { speed_limit[j] = speed; }
        start += distance;
    }

    start = 0;
    // 记录每单位距离的实际速度 单位为 1 mile
    vector<int> real_speed(LEN);
    for (int i = 0; i < M; i++) {
        int length, speed;
        cin >> length >> speed;
        for (int j = start; j < start + length; j++) { real_speed[j] = speed; }
        start += length;
    }
    // 以 1 mile 为单位计算最大超速值
    int ans = 0;
    for (int i = 0; i < LEN; i++) { ans = max(ans, real_speed[i] - speed_limit[i]); }
    cout << ans << endl;
}